sqlmesh/utils/metaprogramming.py

```diff
@@ -276,6 +276,14 @@ def walk(obj: t.Any) -> None:
                 build_env(v, env=env, name=k, path=path)
 
     if name not in env:
+        registry_name = getattr(obj, "registry_name", None)
```

Comment:
can you instead just check that it's a sqlmesh macro?

Reply:
You mean by creating a different attribute like `is_sqlmesh_macro` or something in the decorator's `__call__`?

Reply:
i mean actually checking the name of the macro and seeing that it's macro and that it's path is in sqlmesh?